src/averageAges.js

```diff
@@ -37,7 +55,25 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  if (withChildren) {
+    const womenFilteredChildren = people.filter(person =>
+      person.sex === 'f' && (people.find(item =>
+        item.mother === person.name) !== undefined));
+
+    const ageWomenFilteredChildren = womenFilteredChildren.map(person =>
+      person.died - person.born);
+
+    return ageWomenFilteredChildren.reduce((a, b) =>
+      a + b) / womenFilteredChildren.length;
+  }
+
+  const womenFiltered = people.filter(person =>
+    person.sex === 'f');
+
+  const ageWomenFiltered = womenFiltered.map(person =>
+    person.died - person.born);
+
+  return ageWomenFiltered.reduce((a, b) => a + b) / womenFiltered.length;
```

Comment:
you basically repeat logic of finding averageAge from first function, create a function helper for it